src/lsp/cobol_lsp/lsp_request.ml

```diff
@@ -454,6 +454,39 @@ let handle_semtoks_full,
 
 (** {3 Hover} *)
 
+let get_hover_text (qn: Cobol_ptree.qualname) (cu: Cobol_unit.Types.cobol_unit) =
+  match Cobol_unit.Qualmap.find qn cu.unit_data.data_items.named with
+  | data_def ->
+      Some (Pretty.to_string "%a" Cobol_data.Hover.pp_data_definition data_def)
+  | exception Not_found
+  | exception Cobol_unit.Qualmap.Ambiguous _ ->
+      None
+
+let get_hover_text_and_loc cu_name element_at_pos group =
```

Comment:
You should probably just let `Not_found` escape here and catch it below, in order for `loc` to always be given (and therefore always give a hovered range).
So you can also raise `Not_found` instead of returning `None, None`.

Reply:
Is that what you had in mind ?
https://github.com/OCamlPro/superbol-studio-oss/pull/293/files#diff-6280a7559f43fce76b94b7769d17ead431dcb87c307c61fe4b7eff4b7048fbc1R469-R473